idunn/blocks/images.py

```diff
@@ -0,0 +1,131 @@
+        return cls._salt
+
+    @classmethod
+    def use_thumbr(cls):
+        if cls._use_thumbr is None:
+            from app import settings
+            cls._use_thumbr = settings.get('USE_THUMBR')
+        return cls._use_thumbr
+
+    @classmethod
+    def init_sub_domains(cls):
+        if cls._sub_domains is None:
+            from app import settings
+            cls._sub_domains = settings.get('THUMBR_DOMAINS').split(',')
+
+    @classmethod
+    def get_domain(cls, hash):
+        cls.init_sub_domains()
+        n = int(hashlib.sha256(bytes(hash[0], encoding='utf8')).hexdigest(), 16) % (len(cls._sub_domains))
+        return 'https://' + cls._sub_domains[n] + '/thumbr'
+
+    @classmethod
+    def get_url_remote_thumbnail(cls, source, width=0, height=0, bestFit=True, progressive=False, animated=False):
+        displayErrorImage = False
+
+        salt = cls.get_salt()
+        token = f"{source}{width}x{height}{salt}"
+        hash = hashlib.sha256(bytes(token, encoding='utf8')).hexdigest()
+        domain = cls.get_domain(hash)
+
+        size = int(width) * int(height)
+
+        hashURLpart = f"{hash[0]}/{hash[1]}/{hash[2:]}"
+
+        filename = source.split('/')[-1]
+        if not bool(re.match("^.*\.(jpg|jpeg|png|gif)$", filename)):
+            filename += ".jpg"
+
+        params = f"u={urllib.parse.quote(source)}&q={1 if displayErrorImage else 0}&b={1 if bestFit else 0}&p={1 if progressive else 0}&a={1 if animated else 0}"
+        return domain + "/" + str(size) + "/" + hashURLpart + "/" + filename + "?" + params
+
+class Image(types.Type):
+    url = validators.String()
+    alt = validators.String()
+    credits = validators.String()
+
+class ImagesBlock(BaseBlock):
+    BLOCK_TYPE = "images"
+    images = validators.Array(items=Image)
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        wikidata_id = es_poi.get("properties", {}).get("wikidata")
+        if wikidata_id is not None:
+            wiki_index = WikidataConnector.get_wiki_index(lang)
+            if wiki_index is not None:
+                properties = es_poi.get('properties', {})
+                place_name = properties.get('name')
+                try:
+                    try:
+                        wiki_es = WikidataConnector.get_wiki_es()
+                    except WikiUndefinedException:
+                        logger.info("WIKI_ES variable has not been set: we cannot return any image")
+                        return None
+                    if wiki_es is None:
+                        return None
+                    resp = wiki_es.search(
+                        index=wiki_index,
+                        body={
+                            "filter": {
+                                "term": {
+                                    "wikibase_item": wikidata_id
+                                }
+                            }
+                        }
+                    ).get('hits', {}).get('hits', [])
```

Comment:
We should try not to query the ES a second time, after the wiki block.  
And this implementation should not be duplicated anyway.  
Why not persist the wiki_es response in a wrapper class for es_poi ? Then it could be used by both blocks with a single query.